### PR TITLE
feat: routing api integration

### PR DESCRIPTION
**Pull request by justindomingue**
- add rtk-query api for the routing api
- updates to rtk 1.6.1 for devex

### CODE REVIEW

File: src/hooks/useRouterTradeExactIn.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { skipToken } from '@reduxjs/toolkit/dist/query/react'
```
Comment (moodysalem) [inline src/hooks/useRouterTradeExactIn.ts:1]: is this import always like this? weird

File: src/hooks/useRouterTradeExactIn.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      currencyOut?.chainId &&
+      account &&
+      amountIn !== undefined
+      ? {
+          tokenIn: { address: amountIn.currency.wrapped.address, chainId: amountIn.currency.chainId },
+          tokenOut: {
+            address: currencyOut.wrapped.address,
+            chainId: currencyOut.chainId,
+          },
+          amount: amountIn.toSignificant(6),
```
Comment (moodysalem) [inline src/hooks/useRouterTradeExactIn.ts:25]: `.quotient.toExact()` i'm pretty sure amount is supposed to be without decimals

File: src/hooks/useRouterTradeExactIn.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { useActiveWeb3React } from './web3'
+
+const DEFAULT_SLIPPAGE_TOLERANCE = '5'
+const DEFAULT_DEADLINE = '360'
+
+export function useRouterTradeExactIn(amountIn?: CurrencyAmount<Currency>, currencyOut?: Currency) {
+  const { account } = useActiveWeb3React()
+
+  const { isLoading, isError, data } = useGetQuoteQuery(
+    amountIn?.currency?.wrapped?.address &&
```
Comment (moodysalem) [inline src/hooks/useRouterTradeExactIn.ts:13]: if you just check amountIn the rest should be true

File: yarn.lock [hunk 4]
```diff
@@ -2847,10 +2847,10 @@
     "@react-hook/event" "^1.2.1"
     "@react-hook/throttle" "^2.2.0"
 
-"@reduxjs/toolkit@^1.6.0":
-  version "1.6.0"
-  resolved "https://registry.yarnpkg.com/@reduxjs/toolkit/-/toolkit-1.6.0.tgz#0a17c6941c57341f8b31e982352b495ab69d5add"
-  integrity sha512-eGL50G+Vj5AG5uD0lineb6rRtbs96M8+hxbcwkHpZ8LQcmt0Bm33WyBSnj5AweLkjQ7ZP+KFRDHiLMznljRQ3A==
+"@reduxjs/toolkit@^1.6.1":
+  version "1.6.1"
```
Comment (moodysalem) [inline yarn.lock:2851]: curious why the bump

File: src/hooks/useRouterTradeExactIn.ts (hunk 1 above)
Comment (justindomingue) [reply src/hooks/useRouterTradeExactIn.ts:1]: `dist` was not necessary

File: src/hooks/useRouterTradeExactIn.ts (hunk 2 above)
Comment (justindomingue) [reply src/hooks/useRouterTradeExactIn.ts:25]: @willpote can you confirm?

File: yarn.lock (hunk 4 above)
Comment (justindomingue) [reply yarn.lock:2851]: related to https://github.com/reduxjs/redux-toolkit/issues/1308#event-5034180589

vscode would import `skipToken` from the wrong package which results in a very hard to track issue. They fixed the typing in 1.6.1.

File: src/hooks/useRouterTradeExactIn.ts (hunk 2 above)
Comment (justindomingue) [reply src/hooks/useRouterTradeExactIn.ts]: using  `.quotient.toString()`

File: src/hooks/useRouter.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { skipToken } from '@reduxjs/toolkit/query/react'
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+import { useBlockNumber } from 'state/application/hooks'
+import { useGetQuoteQuery } from 'state/routing/slice'
+import { useActiveWeb3React } from './web3'
+
+const DEFAULT_SLIPPAGE_TOLERANCE = '5'
+const DEFAULT_DEADLINE = '360'
```
Comment (willpote) [inline src/hooks/useRouter.ts:9]: Shouldn't we use the default values the app uses currently for these?

File: src/state/routing/slice.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/dist/query/react'
```
Comment (willpote) [inline src/state/routing/slice.ts:1]: importing from dist

File: src/state/routing/slice.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        tokenOutAddress: string
+        tokenOutChainId: SupportedChainId
+        amount: string
+        type: 'exactIn' | 'exactOut'
+        recipient?: string
+        slippageTolerance?: string
+        deadline?: string
+
+        // internal: invalidates query on block sync
+        blockNumber: number
```
Comment (willpote) [inline src/state/routing/slice.ts]: Discussed offline but this is tricky due to blockNumber sync issues

Comment (justindomingue) [reply src/state/routing/slice.ts]: switched to polling

**Pull request merged**